const {CityService}=require('../services/index')

const cityService=new CityService()

const create= async(req,res)=>{
    try {
        const city=await cityService.createCity(req.body)
        return res.status(201).json({
            data:city,
            success:true,
            message:"successfully city is created ",
            err:{}
        })
        
    } catch (error) {
        console.log("error")
        return res.status(500).json({
            data:{},
            success:false,
            message:'falied to create a city',
            err:error
        })

        
    }

}
//DELETE-> /city/:id
const destroy=async(req,res)=>{
    try {
        const response=await cityService.deleteCity(req.params.id);
        return res.status(200).json({
            data:response,
            success:true,
            message:"successfully city is deleted ",
            err:{}

        })
        
    } catch (error) {
        console.log("error")
        return res.status(500).json({
            data:{},
            success:false,
            message:'falied to delete the  city',
            err:error
        })

        
    }

}
//PUT->/city/:id => req.body
 const update=async(req,res)=>{
    try {
        const response=await cityService.updateCity(req.params.id,req.body);
        return res.status(200).json({
            data:response,
            success:true,
            message:"successfully city is updated ",
            err:{}

        })
        
    } catch (error) {
        console.log("error")
        return res.status(500).json({
            data:{},
            success:false,
            message:'falied to delete the  city',
            err:error
        })
        
    }

 }
 //GET->/city/:id
 const get=async(req,res)=>{
    try {
        const response=await cityService.getCity(req.params.id);
        return res.status(200).json({
            data:response,
            success:true,
            message:"successfully fetched the city ",
            err:{}

        })
        
    } catch (error) {
        console.log("error")
        return res.status(500).json({
            data:{},
            success:false,
            message:'falied to get the  city',
            err:error
        })

        
    }

 }
 module.exports={
    create,
    destroy,
    update,
    get
 }

